server/test/write_authzmodel.go

```diff
@@ -367,37 +550,141 @@ func TestWriteAuthorizationModel(t *testing.T, datastore storage.OpenFGADatastor
 			request: &openfgapb.WriteAuthorizationModelRequest{
-				TypeDefinitions: &openfgapb.TypeDefinitions{
-					TypeDefinitions: []*openfgapb.TypeDefinition{
-						{
-							Type: "foo",
-							Relations: map[string]*openfgapb.Userset{
+				SchemaVersion: "1.1",
+				StoreId:       storeID,
+				TypeDefinitions: []*openfgapb.TypeDefinition{
+					{
+						Type: "user",
+					},
+					{
+						Type: "group",
+						Relations: map[string]*openfgapb.Userset{
+							"admin":  {Userset: &openfgapb.Userset_This{}},
+							"member": {Userset: &openfgapb.Userset_This{}},
+						},
+						Metadata: &openfgapb.Metadata{
+							Relations: map[string]*openfgapb.RelationalMetadata{
+								"admin": {
+									DirectlyRelatedUserTypes: []*openfgapb.RelationReference{
+										{
+											Type: "user",
+										},
+									},
+								},
+								"member": {
+									DirectlyRelatedUserTypes: []*openfgapb.RelationReference{
+										{
+											Type: "user",
+										},
+									},
+								},
+							},
+						},
+					},
+					{
+						Type: "document",
+						Relations: map[string]*openfgapb.Userset{
+							"reader": {Userset: &openfgapb.Userset_This{}},
+							"writer": {Userset: &openfgapb.Userset_This{}},
+						},
+						Metadata: &openfgapb.Metadata{
+							Relations: map[string]*openfgapb.RelationalMetadata{
+								"reader": {
+									DirectlyRelatedUserTypes: []*openfgapb.RelationReference{
+										{
+											Type: "user",
+										},
+										{
+											Type:          "group",
+											RelationOrAny: &openfgapb.RelationReference_Relation{Relation: "member"},
+										},
+									},
+								},
 								"writer": {
-									Userset: &openfgapb.Userset_ComputedUserset{
-										ComputedUserset: &openfgapb.ObjectRelation{
-											Object:   "",
-											Relation: "reader",
+									DirectlyRelatedUserTypes: []*openfgapb.RelationReference{
+										{
+											Type: "user",
+										},
+										{
+											Type:          "group",
+											RelationOrAny: &openfgapb.RelationReference_Relation{Relation: "admin"},
 										},
 									},
 								},
+							},
+						},
+					},
+				},
+			},
+		},
+		{
+			name: "unsupported schema version",
+			request: &openfgapb.WriteAuthorizationModelRequest{
+				SchemaVersion: "xxx.1",
+				StoreId:       storeID,
+				TypeDefinitions: []*openfgapb.TypeDefinition{
+					{
+						Type: "document",
+					},
+				},
+			},
+			err: errors.UnsupportedSchemaVersion,
+		},
+		{
+			name: "relational type which doesn't exist",
+			request: &openfgapb.WriteAuthorizationModelRequest{
+				SchemaVersion: "1.1",
+				StoreId:       storeID,
+				TypeDefinitions: []*openfgapb.TypeDefinition{
+					{
+						Type: "document",
+						Relations: map[string]*openfgapb.Userset{
+							"reader": {Userset: &openfgapb.Userset_This{}},
+						},
+						Metadata: &openfgapb.Metadata{
+							Relations: map[string]*openfgapb.RelationalMetadata{
 								"reader": {
```

Comment:
can you add a test for the case where the auth model has 2 types but only one has defined metadata? (we should not allow it)